unblob/handlers/executable/elf.py

```diff
@@ -26,6 +28,27 @@
 KERNEL_MODULE_SIGNATURE_FOOTER = b"~Module signature appended~\n"
 
 
+@attr.define(repr=False)
+class ElfChunk(ValidChunk):
+    def extract(self, inpath: Path, outdir: Path):
+        # ELF file extraction is special in that in the general case no new files are extracted, thus
+        # when we want to clean up all carves to save place, carved ELF files would be deleted as well,
+        # however we want to keep carved out ELF files, as they are the interesting stuff!
+        if self.is_whole_file:
+            # whole file chunks are not carved, so we can do normal extraction
+            super().extract(inpath, outdir)
+        else:
+            # make a copy, and let the carved chunk be deleted
+            outdir.mkdir(parents=True, exist_ok=False)
```

Comment:
wondering if we can simplify it even further, considering the options:

- 1: simple ELF file, whole file -> nothing to do, we already have the ELF file
- 2: simple ELF file, not whole file -> need to copy the carve to have an ELF file
- 3: kernel ELF part, whole file -> need to extract the initramfs part, no need to do anything with the ELF file
- 4 kernel ELF part, not whole file -> need to extract the initramfs, also no need to anything with the ELF part as the kernel image might contain an ELF part but we already have the kernel image as a parent file

Reply:
I do not get why the last item is valid, and without that one, the first 3 describes the current MR behavior.

One thing I have in mind is a disk image which is not of a known filesystem format or damaged.
While we can carve the ELF kernel, and extract the initramfs, the kernel file itself will not be kept.

Reply:
OK, so the kernel is not an ELF by itself, just part of it, and that part is not that important by itself, just the initramfs, so the ELF can be deleted - if needed the chunk information and the kernel image is there.